File: N3.15/N3.15.java
import java.util.Scanner;

public class NumberReverser {
    public static boolean reglamentCheck(int num) {
        int prevDigit = num % 10;
        num = num / 10;
        while (num != 0) {
            int currentDigit = num % 10;
            if (currentDigit <= prevDigit) {
                prevDigit = currentDigit;
                num /= 10;
            } else {
                return false;
            }
        }
        return true;
    }

    public static int numReverser(int numBlank) {
        if (reglamentCheck(numBlank)) {
            int reversedNum = 0;
            while (numBlank != 0) {
                reversedNum = reversedNum * 10 + numBlank % 10;
                numBlank /= 10;
            }
            return reversedNum;
        } else {
            return 0;
        }
    }

    public static void main(String[] args) {
        final int defaultCount = 4;
        int n = defaultCount;

        Scanner scanner = new Scanner(System.in);
        for (int i = 0; i < n; ++i) {
            int number = scanner.nextInt();
            System.out.print(numReverser(number) + " ");
        }
        System.out.println();
    }
}

File: N3.15/N3.15.rb
def reglament_check(num)
    prev_digit = num % 10
    num = num / 10
    while num != 0
      current_digit = num % 10
      if current_digit <= prev_digit
        prev_digit = current_digit
        num /= 10
      else
        return false
      end
    end
    true
  end
  
  def num_reverser(num_blank)
    if reglament_check(num_blank)
      reversed_num = 0
      while num_blank != 0
        reversed_num = reversed_num * 10 + num_blank % 10
        num_blank /= 10
      end
      reversed_num
    else
      0
    end
  end
  
  # Define the numbers you want to process
  numbers = [1234, 4321, 2111, 9876]
  
  numbers.each do |number|
    puts "#{num_reverser(number)} "
  end
  puts